web/src/components/AssertionForm/DurationFields.tsx

```diff
@@ -0,0 +1,61 @@
+import {FormInstance, Input, Select} from 'antd';
+import {TAssertion} from '../../types/Assertion.types';
+import {IValues} from './AssertionForm';
+import * as S from './AssertionForm.styled';
+
+type Duration = 'ms' | 's';
+
+function setDurationFieldValue(
+  form: FormInstance<IValues>,
+  index: number,
+  state: {value?: string; duration?: Duration}
+) {
+  const fields = form.getFieldsValue();
+  form.setFieldsValue({
+    assertionList: fields.assertionList?.map((as, i) =>
+      i === index ? {...as, expected: `${state.value}${state.duration}`} : as
+    ),
+  });
+}
+
+interface IProps {
+  assertion: TAssertion;
+  form: FormInstance<IValues>;
+  index: number;
+}
+
+export const DurationFields = ({form, index, assertion}: IProps) => {
+  const state = {
+    value: assertion?.expected?.match(/(\d+)/)?.[0],
+    duration: (assertion?.expected?.match(/[a-zA-Z]+/g)?.[0] || 'ms') as Duration,
+  };
+  return (
+    <div style={{display: 'flex'}}>
+      <Input
+        value={state.value}
+        onChange={e => setDurationFieldValue(form, index, {duration: state.duration, value: e.target.value})}
+        placeholder="Expected Value"
+        type="number"
+      />
+      <S.Select
+        onChange={e => setDurationFieldValue(form, index, {value: state.value, duration: e as Duration})}
+        value={state.duration}
+        style={{margin: 0}}
+        placeholder="Assertion Type"
+      >
+        {[
+          {label: 'nanoseconds', value: 'ns'},
```

Comment:
Let's move this to a constant

Reply:
created an enum and created this array based on that